functionalDB/chapter.md

```diff
@@ -0,0 +1,1098 @@
+(defn remove-entity [db ent-id]
+   (let [ent (entity-at db ent-id)
+         layer (remove-back-refs db ent-id (last (:layers db)))
+         no-ref-layer (update-in layer [:VAET] dissoc ent-id)
+         no-ent-layer (assoc no-ref-layer :storage 
+                                   (drop-entity  
+                                          (:storage no-ref-layer) ent))
+         new-layer (reduce (partial remove-entity-from-index ent) 
+                                 no-ent-layer (indices))]
+     (assoc db :layers (conj  (:layers db) new-layer))))
+````
+The reference removal part is handled in the *remove-back-refs* function.
+
+````clojure
+(defn- remove-back-refs [db e-id layer]
+   (let [refing-datoms (reffing-datoms-to e-id layer)
+         remove-fn (fn[d [e a v]] (update-datom db e a v :db/remove))
+         clean-db (reduce remove-fn db refing-datoms)]
+     (last (:layers clean-db))))
+````
+It starts by locating the entities that refer to the removed entity, by calling *reffing-datoms-to* that returns a sequence of triplets, each containing the id of the referencing entity, the attribute name and the id of the removed entity.
+
+````clojure
+(defn- reffing-datoms-to [e-id layer]
+   (let [vaet (:VAET layer)]
+         (for [[attr-name reffing-set] (e-id vaet)
+               reffing reffing-set]
+              [reffing attr-name e-id])))
+
+````
+ When the *remove-back-refs* function gets the triplets sequence, it just calls the *update-datom* function (which will be explain in the next part) to update the referencing entity to not hold the id of the removed entity (at the found attribute).
+
+The reference removal process ends with clearing-up the removed entity’s id from the VAET index (the index that holds the references) at the *remove-entity* function.
+
+#### Updating an entity
+
+At its essence, an update is the modification of an entity’s attribute’s value. The modification itself depends on the cardinality of the attribute: an attribute whose cardinality is *:db/multiple* (meaning that the attribute holds a set of values) allows adding or removing items to/from it, or replacing the entire set with a new set. An attribute whose cardinality is *:db/single* (meaning that the attribute holds a single value) allows only to replace the value by another one.  
+
+Any modification of a value also needs to ripple throughout the database by modifying the various indices to reflect the change that occurred. 
+
+At this point It is important to keep in mind that the meaning of modifying here is not overwriting, but rather adding an additional layer that holds the updated database state. That updated attribute’s value in the new layer should point to the pre-update value, so we could track the changes in an entity throughout time.
+
+The update process is done at the function *update-datom*. It creates the new attribute and updates the relevant indices and ends with the creation of a new layer.
+
+````clojure
+(defn update-datom

```

Comment:
After reading through this a few times, I think it might be better to call this function `update-entity`, and then to explain that we use a datom as input to this function because it's how we describe the update? This would maintain naming consistency across the "low-level manipulation" API.